extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/RelationalJdbcConfiguration.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import io.quarkus.runtime.annotations.StaticInitSafe;
+import io.smallrye.config.ConfigMapping;
+import java.util.Optional;
+
+@StaticInitSafe
+@ConfigMapping(prefix = "polaris.persistence.relational")
+public interface RelationalJdbcConfiguration {
```

Comment:
Sorry, I should have commented about Quarkus support for DataSources before (see my other comment). If that works, all this config can be delegated to Quarkus, I think.... Essentially this PR would become redundant... Why not repurpose this PR and put all of `relational-jdbc` code here?

Reply:
> Why not repurpose this PR and put all of relational-jdbc code here?

That does makes sense, should be roughly ~1.5K LOC, let me do that !